ast/src/asm_analysis/display.rs

```diff
@@ -47,23 +48,32 @@ where
 
 impl<T: Display> Display for Machine<T> {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        match (&self.latch, &self.function_id) {
-            (Some(latch), Some(function_id)) => write!(f, "({latch}, {function_id})"),
+        match (&self.latch, &self.operation_id) {
+            (Some(latch), Some(operation_id)) => write!(f, "({latch}, {operation_id})"),
             (None, None) => write!(f, ""),
             (Some(latch), None) => write!(f, "({latch}, _)"),
-            (None, Some(function_id)) => write!(f, "(_, {function_id})"),
+            (None, Some(operation_id)) => write!(f, "(_, {operation_id})"),
         }?;
 
         writeln!(f, " {{")?;
+
         write_indented_items(f, &self.degree)?;
         write_indented_items(f, &self.registers)?;
         write_indented_items(f, &self.instructions)?;
-        write_indented_items(f, &self.functions)?;
+        write_indented_items(f, &self.callable)?;
         write_indented_items(f, &self.constraints)?;
+        write_indented_items(f, &self.links)?;
+
         writeln!(f, "}}")
     }
 }
 
+impl<T: Display> Display for LinkDefinitionStatement<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        write!(f, "link {} {} = {};", self.flag, self.params, self.to)
```

Comment:
doesn't it have the arrow too?

Reply:
The arrow is included in the params : inputs -> outputs